### ISSUE TITLE
CNN training on translation task gives `AttributeError: 'MMapIndexedDataset' object has no attribute '_bin_buffer_mmap'`

### ISSUE THREAD

**Issue by HieuFromWaterloo** [pytorch/fairseq#3713]
#### What is your question?
I followed the instructions to train the fconv_wmt_en_de model  and got the following error.

```
Traceback (most recent call last):
  File "/home/ssm-user/anaconda3/envs/venv2/bin/fairseq-train", line 33, in <module>
    sys.exit(load_entry_point('fairseq', 'console_scripts', 'fairseq-train')())
  File "/home/ssm-user/auco_fairseq/fairseq/fairseq_cli/train.py", line 359, in cli_main
    main(args)
  File "/home/ssm-user/auco_fairseq/fairseq/fairseq_cli/train.py", line 68, in main
    task.load_dataset(valid_sub_split, combine=False, epoch=1)
  File "/home/ssm-user/auco_fairseq/fairseq/fairseq/tasks/translation.py", line 257, in load_dataset
    truncate_source=self.args.truncate_source,
  File "/home/ssm-user/auco_fairseq/fairseq/fairseq/tasks/translation.py", line 77, in load_langpair_dataset
    tgt_dataset = data_utils.load_indexed_dataset(prefix + tgt, tgt_dict, dataset_impl)
  File "/home/ssm-user/auco_fairseq/fairseq/fairseq/data/data_utils.py", line 80, in load_indexed_dataset
    dictionary=dictionary,
  File "/home/ssm-user/auco_fairseq/fairseq/fairseq/data/indexed_dataset.py", line 60, in make_dataset
    return MMapIndexedDataset(path)
  File "/home/ssm-user/auco_fairseq/fairseq/fairseq/data/indexed_dataset.py", line 448, in __init__
    self._do_init(path)
  File "/home/ssm-user/auco_fairseq/fairseq/fairseq/data/indexed_dataset.py", line 458, in _do_init
    self._index = self.Index(index_file_path(self._path))
  File "/home/ssm-user/auco_fairseq/fairseq/fairseq/data/indexed_dataset.py", line 408, in __init__
    self._dtype = dtypes[dtype_code]
KeyError: 9
Exception ignored in: <bound method MMapIndexedDataset.Index.__del__ of <fairseq.data.indexed_dataset.MMapIndexedDataset.Index object at 0x7fe9165e2da0>>
Traceback (most recent call last):
  File "/home/ssm-user/auco_fairseq/fairseq/fairseq/data/indexed_dataset.py", line 423, in __del__
    self._bin_buffer_mmap._mmap.close()
AttributeError: 'Index' object has no attribute '_bin_buffer_mmap'
Exception ignored in: <bound method MMapIndexedDataset.__del__ of <fairseq.data.indexed_dataset.MMapIndexedDataset object at 0x7fe9165e2c88>>
Traceback (most recent call last):
  File "/home/ssm-user/auco_fairseq/fairseq/fairseq/data/indexed_dataset.py", line 465, in __del__
    self._bin_buffer_mmap._mmap.close()
AttributeError: 'MMapIndexedDataset' object has no attribute '_bin_buffer_mmap'
```

#### Code

GPU: NVIDIA Tesla T4

```
CUDA_VISIBLE_DEVICES=0 fairseq-train \
    data-bin/fconv \
    --arch fconv_wmt_en_de \
    --encoder-embed-dim 128 \
    --decoder-embed-dim 128 \
    --max-tokens 6000 \
    --optimizer adam \
    --lr 0.25 \
    --lr-scheduler fixed \
    --force-anneal 50 \
    --clip-norm 0.1 \
    --dropout 0.2 \
    --save-interval 1 --max-epoch 20 \
    --criterion label_smoothed_cross_entropy --label-smoothing 0.1 \
    --skip-invalid-size-inputs-valid-test \
    --patience 3
```

#### What have you tried?

- I looked and tried this: https://github.com/pytorch/fairseq/issues/1903 (even though it's not the same problem)
- I also tried to different pytorch version: 1.8, 1.7, 1.5
- I ran `transformer_iwslt_de_en` on the same dataset without any error
- No errors when I tried to reduce the vocabulary size from 68000 to 50000. I wonder why this is the case?

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): master
 - PyTorch Version (e.g., 1.0): 1.5.0+cu101 (also tried 1.8.0+cu111)
 - OS (e.g., Linux): Linux
 - How you installed fairseq (`pip`, source): source
 - Build command you used (if compiling from source): pip install --editable .
 - Python version: 3.6.13
 - CUDA/cuDNN version: 11.0
 - GPU models and configuration: NVIDIA Tesla T4

Thank you!

**Comment by nikhiljaiswal** [comment]
hi did you resolve this error?

**Comment by HieuFromWaterloo** [comment]
i resolved this issues by using python 3.6.8 and pytorch1.5.0.
